### PR TITLE
Add Segment.screen and Segment.screenWithProperties

### PR DESCRIPTION
**Pull request by dozoisch**
- Unified the .track on java to have parameter eventName for both track and trackWithProperties function
- Added screen methods. Based on track events for naming conventions

---

I'm not exactly sure how to test this when setup to run on expo. I based implementation on the documentation itself.

### CODE REVIEW

File: android/expoview/src/main/java/versioned/host/exp/exponent/modules/api/SegmentModule.java [hunk 1]
```diff
@@ -100,9 +100,9 @@ public void identifyWithTraits(final String userId, final ReadableMap properties
   }
 
   @ReactMethod
-  public void track(final String event) {
+  public void track(final String eventName) {
     if (mClient != null) {
-      mClient.track(event);
+      mClient.track(eventName);
```
Comment (dozoisch) [inline android/expoview/src/main/java/versioned/host/exp/exponent/modules/api/SegmentModule.java]: trackWithProperties had eventName so made both have similar param names

**Comment by dozoisch** [comment]
hello @skevy @jesseruder anything missing to get this merged?

**Review by jesseruder — COMMENTED** [review]
Looks good on android side! @terribleben does ios look fine?

**Comment by dozoisch** [comment]
@terribleben @jesseruder any update on this? Would be nice to have this in next sdk version? As it's a big blocker for us. 

Anything I can do to help? I am willing to update the docs if someone can point me where to do it!

**Comment by abi** [comment]
@dozoisch Update docs here: https://github.com/expo/expo-docs/blob/master/versions/unversioned/sdk/segment.md thanks!

**Comment by terribleben** [comment]
Hi, really sorry, I just didn't notice the notifications for this PR amidst the millions of other notifications.

We're releasing SDK 19 today-ish (obviously without this), but we can probably pull this into a minor release soonish.

**Comment by dozoisch** [comment]
@terribleben no worries, completely understands that!

I'll put up a PR for the documentation updates.

**Pull request closed without merging**